blog/book-review-the-goal-a-process-of-ongoing-improvement.org

```diff
@@ -0,0 +1,229 @@
+  + throughput ::
+    the rate at which the system generates money through sales.
+    (the money coming in)
+  + inventory ::
+    all the money that the system has invested in purchasing things which it intends to sell.
+    (the money currently inside the system)
+  + operational expense ::
+    all the money the system spends in order to turn inventory into throughput.
+    (the money we have to pay out to make throughput happen)
+  #+END_QUOTE
+
+** How to making more money with software?
+So what metrics can we use to measure the productivity of software development?
+Again: Throughput, Inventory, and Operational Expense.
+Because software development is production as well. (See also [[id:A86C142A-B9ED-4242-A4BC-77B3571C835E][The Phoenix Project]])
+So what exactly does these three metrics mean in software development?
+- throughput ::
+  This one is easy.
+  If you are a SaaS company with a subscription model, your throughput is how much subscription fee you get from your users.
+- inventory ::
+  This one is tricky.
+
+  I think inventory in software development as the features that are under development and are still not shipped to users.
+  And you can see inventory at every step of software development:
+  1. Feature requirements / specifications that are not runnable code yet.
+  2. Design mockups / wireframes that are not usable at all.
+  3. Code that are not merged nor deployed to production.
+  As the book explains, *when inventory goes up, the carrying costs goes up*.
+  The more feature branches you have, the more integration costs you need to pay.
+  The more features you are working on, the more possible to ship something that is not working as expected.
+  This is why we need to limit WIP. Because it helps reduce our inventory.
+
+  But the question then becomes how to measure the value of these WIP features?
+  It's a hard question and the answer varies from product to product.
+  (We may use how much a user want to pay for it and how many users may pay for it to estimate the value of a WIP feature.)
+- operational expense ::
+  Everything else is operational expense.
+  + Labor costs.
+  + Tools we are using (Laptops, editors, IDEs, software licenses, etc).
+  + The server that our code runs on.
+
+Only by understanding these 3 key metrics, can we start identifying the bottleneck in our feature development pipeline.
+** How to making more money as an individual?
+We can apply the same thinking to an individual as well:
+If our goal as an individual is to make money, then what's our throughput, inventory, and operational expense?
+- throughput ::
+  If you are like me, having a stable work and income, our throughput is very stable and predictable: salary, compensations, etc.
+- inventory ::
+  Then, what is our inventory?
+  What's the money currently inside us?
+
+  If we think at a higher level, our inventory is the knowledge we have.
```

Comment:
Interesting. I think the difference between a human being and a company is that, there is usually no obivous downside or "carrying cost" to have more knowledge.